core/domain/stats_jobs_one_off.py

```diff
@@ -16,16 +16,524 @@
+            for state_name in state_hit_counts_for_this_version:
+                state_stats_mapping[state_name].total_hit_count_v1 += (
+                    state_hit_counts_for_this_version[state_name][
+                        'total_hit_count'])
+                state_stats_mapping[state_name].first_hit_count_v1 += (
+                    len(state_session_ids_by_version[version][state_name]))
+
+            # Compute num_completions for the states.
+            for state_name in state_completion_counts_for_this_version:
+                state_stats_mapping[state_name].num_completions_v1 += (
+                    state_completion_counts_for_this_version[state_name])
+
+            # Compute total_answers_count and useful_feedback_count.
+            for state_name in state_answer_counts_for_this_version:
+                state_stats_mapping[state_name].total_answers_count_v1 += (
+                    state_answer_counts_for_this_version[state_name][
+                        'total_answers_count'])
+                state_stats_mapping[state_name].useful_feedback_count_v1 += (
+                    state_answer_counts_for_this_version[state_name][
+                        'useful_feedback_count'])
+
+            # If the exploration has only one state, then the number of actual
+            # starts will be equal to the number of starts.
+            if len(versioned_exploration.states) == 1:
+                num_actual_starts = num_starts
+            else:
+                # Now that the stats for the exploration's version are computed,
+                # calculate num_actual_starts. To compute this, we take the max
+                # of the first hit counts of all states that the initial state
+                # leads to. The max is taken instead of sum because the sum of
+                # first_hit_counts will sometimes end up counting a single
+                # learner twice. The max of both first hit counts ensures that
+                # we arecounting only unique learners who traversed past the
+                # initial state.
+                init_state = versioned_exploration.states[
+                    versioned_exploration.init_state_name]
+                max_first_hit_from_init_state = max([
+                    state_stats_mapping[
+                        answer_group.outcome.dest].first_hit_count_v1
+                    for answer_group in init_state.interaction.answer_groups
+                    if answer_group.outcome.dest != (
+                        versioned_exploration.init_state_name)] or [0])
+                num_actual_starts = max_first_hit_from_init_state
+
+            # Check if model already exists. If it does, update it, otherwise
+            # create a fresh ExplorationStatsModel instance.
+            exploration_stats = exploration_stats_by_version[version - 1]
+            if exploration_stats is not None:
+                exploration_stats = stats_domain.ExplorationStats(
```

Comment:
Do you need to store this back in the exploration_stats_by_version list (in order to handle later reverts)?

Reply:
@seanlip Yep, this needs to.